version: '3.9'

services:
  postgresql:
    image: postgis/postgis:${POSTGRES_VERSION}
    container_name: vallengeo_db
    restart: "on-failure"
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_HOST_AUTH_METHOD: trust
      POSTGRES_MULTIPLE_DATABASES: ${POSTGRES_MULTIPLE_DATABASES}
      POSTGRES_MULTIPLE_EXTENSIONS: postgis,postgis_tiger_geocoder,postgis_topology,fuzzystrmatch,uuid-ossp,unaccent
      PGDATA: /var/lib/postgresql/data
    volumes:
    - vallengeo_db_data/:/var/lib/postgresql/data/
    - ./init-db.sh:/docker-entrypoint-initdb.d/init-db.sh
    networks:
    - vallenge
    ports:
    - ${POSTGRES_DB_DOCKER_PORT}:${POSTGRES_DB_LOCAL_PORT}

volumes:
  vallengeo_db_data:

networks:
  vallenge:
  